### DISCUSSION TITLE
Support for PSP?

### DISCUSSION THREAD

**Discussion by vKrazy** [junguler/m3u-radio-music-playlists#28] [General]
PSP Radio Stations are just m3u lists over an HTML trough the Browser. Mostly supported by ShoutCast and IceCast.org.
There's a way to include support for PSP from this? I can provide the Radio Stations SDK over Discord if that can help create new custom PSP Stations (shorcuts).

**Comment by junguler** [comment]
i'm interested to support as many devices as possible but haven't really researched how it's done on psp 

if all it takes is having icecast and/or shoutcast only m3u files it's something that can be easily done but if it's more involved i need to look into it further

what would be really helpful is you testing what is already here, especially the [icecast](https://github.com/junguler/m3u-radio-music-playlists/tree/main/icecast) folder, does it work out of the box? if not what needs to be done it to make it work, preferably something that can be done on the linux command line so i can do it in bulk after figuring out what operations are needed

**Comment by junguler** [comment]
oh and btw, about shoutcast, unlike icecast i haven't really found a website dedicated to shoutcast only streams, so unless the stream has the word shoutcast in it's address i'm not able to find it

**Comment by vKrazy** [comment]
All the information researched by the community, including all the PSP Stations and the SDK is here. I highly recommend take a look and read the post.
https://www.reddit.com/r/PSP/comments/12gsq4m/psp_internet_radio_in_2023/?rdt=39897

**Comment by junguler** [comment]
ok, i've bookmarked that reddit page, will look into it in a few days when i have more time.

**Comment by junguler** [comment]
hello again, i've checked the google drive archive from that reddit page, the `.prs` files seem to be in 2 letter hex format, i don't really know how to decode them or see what they are made of

your best bet is to either find an audio player for psp that plays online streams or record the streams in your other devices and put them on your psp and play them locally

tldr: i don't know how i would go about doing what you want so please find and alternative way of enjoying radio on your psp

**Comment by vKrazy** [comment]
But have you take a look in the PDF inside the SDK Zip? There explain how to make the PRS, what are made of and the tool used to create the PRS itself

**Comment by junguler** [comment]
i see, no i haven't looked at the sdk zip, ok i guess, give me about a day and i see what's going on, no promises tho

**Comment by junguler** [comment]
hello again, i've taken a look at that pdf file and if i'm understanding things correctly i need to make an html file for each stream? that's too much work for such an extreme edge case like this, it's simply not worth it to me to take the time to understand and actually do this

i'm sorry about taking long to answer and if you had hoped it might work out, please use another device for your radio enjoying needs or like i said record the streams and side-load them to your psp

**Comment by vKrazy** [comment]
> hello again, i've taken a look at that pdf file and if i'm understanding things correctly i need to make an html file for each stream? 

Basically, it works like your repository to listen radio trough the web browser. You can create a composition of radio categories with X amount of radios
![image](https://github.com/junguler/m3u-radio-music-playlists/assets/63470108/319d0e02-ee9b-4d8a-a6fa-2dbf005a78c2)

Or create a simple focused themed radio which changes the station pushing the button
![image](https://github.com/junguler/m3u-radio-music-playlists/assets/63470108/a512bd97-5cc5-4592-908a-b00a93f2ff22)

The PRS is just like a Desktop shorcut, which include mandatory "Radio Player Name" and "URL", then, optionals, such as Icon Image, focus Image, comment about the Internet Radio Player, Author WebSite URL,  Author Name and Copyright Information. 
![image](https://github.com/junguler/m3u-radio-music-playlists/assets/63470108/fba2072b-0879-40f2-8b33-c86a9ac3d54f)


The radio player is placed on a Web server provided by the Internet radio player's author. The radio player is a client-side application that runs on the browser. It is not a server-side application. Therefore, server-side scripts are not necessary.
Besides, part of the processing of an Internet radio player can also be implemented by a server-side cgi program or service such as cron.

To receive an audio stream, the radio player connects to the broadcast server either by directly specifying its URL, or from information contained in playlist or M3U data that points to the desired server.

An ordinary Web page alone cannot function as an Internet radio player. Mechanisms for producing sound and for finding broadcast stations are required for it to function as an Internet radio player. To implement these capabilities on an ordinary Web page, 59 JavaScript extended methods are provided as part of the PSP Internet radio system. These extended methods will give the Web page the additional capabilities needed to function as an Internet radio player. When the Internet browser is launched as a PSP Internet radio player from the Internet radio icon under the "Network" category of the XMB, this group of 52 extended methods is added to the PSP Internet browser. When the browser is started up as a normal PSP Internet browser, this group of methods cannot be used.

It is technically possible to receive a live audio data stream by invoking the action script "Sound.loadSound()" on the Flash player within the PSP Internet browser, however, proper operation is not currently guaranteed. This is because the Flash player that is currently implemented in system software will continuously accumulate a received audio data stream in the PSP system's internal memory. Therefore, playing (receiving) audio using the Flash player's "Sound.loadSound()" will eventually cause an "Insufficient memory" alert dialog to appear.

**Comment by junguler** [comment]
it seems you already know way more than what i ever hope to know, so why not do this yourself?
